shell/common/snapshot_controller_impeller.cc

```diff
@@ -70,6 +56,62 @@ sk_sp<DlImage> SnapshotControllerImpeller::DoMakeRasterSnapshot(
   return nullptr;
 }
 
+sk_sp<DlImage> DoMakeRasterSnapshot(
+    sk_sp<DisplayList> display_list,
+    SkISize picture_size,
+    const std::shared_ptr<const fml::SyncSwitch>& sync_switch,
+    const std::shared_ptr<impeller::AiksContext>& context) {
+  sk_sp<DlImage> result;
+  sync_switch->Execute(fml::SyncSwitch::Handlers()
+                           .SetIfTrue([&] {
+                             // Do nothing.
+                           })
+                           .SetIfFalse([&] {
+                             result = DoMakeRasterSnapshot(
+                                 display_list, picture_size, context);
+                           }));
+
+  return result;
+}
+}  // namespace
+
+void SnapshotControllerImpeller::MakeRasterSnapshot(
+    sk_sp<DisplayList> display_list,
+    SkISize picture_size,
+    std::function<void(const sk_sp<DlImage>&)> callback) {
+  sk_sp<DlImage> result;
+  std::shared_ptr<const fml::SyncSwitch> sync_switch =
+      GetDelegate().GetIsGpuDisabledSyncSwitch();
+  sync_switch->Execute(
+      fml::SyncSwitch::Handlers()
+          .SetIfTrue([&] {
+            std::shared_ptr<impeller::AiksContext> context =
+                GetDelegate().GetAiksContext();
+            if (context) {
+              context->GetContext()->StoreTaskForGPU(
+                  [context, sync_switch, display_list = std::move(display_list),
```

Comment:
I tried to store a weak_ptr to the context here and it didn't work.  It looks like AiksContext doesn't have the proper lifespan.

Reply:
Can you be a bit more specific? Does the AiksContext die during backgrounding?

Reply:
No, it doesn't have to do with backgrounding.  They don't seem to have a lifespan that is as long lived as the impeller context.  I didn't look into what their lifespan was.  I decided practically it was fine to have a strong retain on it while we don't have the GPU since that means we will not be executing anyways so the idea that the `toImage` result will no longer will be relevant when the GPU comes back is low.

Reply:
SG